docs/docs/integrations/retrievers/azure_ai_search.ipynb

```diff
@@ -0,0 +1,292 @@
+    ">Search is foundational to any app that surfaces text to users, where common scenarios include catalog or document search, online retail apps, or data exploration over proprietary content. When you create a search service, you'll work with the following capabilities:\n",
+    "> - A search engine for full text search over a search index containing user-owned content\n",
+    "> - Rich indexing, with lexical analysis and optional AI enrichment for content extraction and transformation\n",
+    "> - Rich query syntax for text search, fuzzy search, autocomplete, geo-search and more\n",
+    "> - Programmability through REST APIs and client libraries in Azure SDKs\n",
+    "> - Azure integration at the data layer, machine learning layer, and AI (AI Services)\n",
+    "\n",
+    "This notebook shows how to use Azure AI Search (AAS) within LangChain."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "074b0004",
+   "metadata": {},
+   "source": [
+    "## Set up Azure AI Search\n",
+    "\n",
+    "To set up AAS, please follow the instructions [here](https://learn.microsoft.com/en-us/azure/search/search-create-service-portal).\n",
+    "\n",
+    "Please note you will need:\n",
+    "\n",
+    "1. the name of your AAS service, \n",
+    "2. the name of your AAS index,\n",
+    "3. your API key.\n",
+    "\n",
+    "Your API key can be either Admin or Query key, but as we only read data it is recommended to use a Query key."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0474661d",
+   "metadata": {},
+   "source": [
+    "## Using the Azure AI Search Retriever"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "39d6074e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "from langchain_community.retrievers import (\n",
+    "    AzureAISearchRetriever,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b7243e6d",
+   "metadata": {},
+   "source": [
+    "Set Service Name, Index Name and API key as environment variables (alternatively, you can pass them as arguments to `AzureAISearchRetriever`). The search index name you use determines which documents are queried, so be sure to select the right one. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "33fd23d1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "os.environ[\"AZURE_AI_SEARCH_SERVICE_NAME\"] = \"<YOUR_AAS_SERVICE_NAME>\"\n",
+    "os.environ[\"AZURE_AI_SEARCH_INDEX_NAME\"] = \"<YOUR_AAS_INDEX_NAME>\"\n",
+    "os.environ[\"AZURE_AI_SEARCH_API_KEY\"] = \"<YOUR_API_KEY>\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "057deaad",
+   "metadata": {},
+   "source": [
+    "Create the Retriever\n",
+    "\n",
+    "`content_key` is the key in the retrieved result to set as the Document page_content.\n",
+    "`top_k` is the number of number of results you'd like to retrieve. Setting it to None (the default) returns all results. "
```

Comment:
There are 2 different parameters right now. top and k for vector queries. Does top_k set one or both of these?

Reply:
The current code is setting only the top parameter. Do you think it's worth it to set both?

`top_param = f"&$top={self.top_k}" if self.top_k else ""`